### ISSUE TITLE
ContextVar not propagated from fixture to test

### ISSUE THREAD

**Issue by andredias** [delfick/alt-pytest-asyncio#8]
As `pytest-asyncio`, it seems that `alt-pytest-asyncio` also doesn't propagate ContextVar to tests. Instead of duplicating the problem description, I will list the references to the issue below:

* [ContextVar not propagated from fixture to test #127](https://github.com/pytest-dev/pytest-asyncio/issues/127)
* [Fix contextvars not propagated from fixture to test #161](https://github.com/pytest-dev/pytest-asyncio/pull/161)
* [How To Write Asynchronous Code With Contextvars Properly](https://www.linw1995.com/en/blog/How-To-Write-Asynchronous-Code-With-Contextvars-Properly/)

**Comment by andredias** [comment]
I've got a new issue on alt-pytest-asyncio that might be related to #8 since the library [databases relies on ContextVars](https://github.com/encode/databases/blob/ae3fb16f40201d9ed0ed31bea31a289127169568/databases/core.py#L39). I created a minimal example to show the problem:

```python
from collections.abc import AsyncIterable

from databases import Database
from pytest import fixture


@fixture
async def db() -> AsyncIterable[Database]:
    db: Database = Database("sqlite:///", force_rollback=True)
    await db.connect()
    try:
        yield db
    finally:
        await db.disconnect()


async def test_db(db: Database) -> None: ...
```

It is just one `test_1.py` file that is supposed to be run by `pytest test_issue.py`. It is supposed to run flawlessly but I get this error:

```bash
$ pytest
=================================== test session starts ===================================
platform linux -- Python 3.12.3, pytest-8.2.0, pluggy-1.5.0
rootdir: /tmp/databases
plugins: alt-pytest-asyncio-0.7.2
collected 1 item                                                                          

tests/test_1.py .E                                                                  [100%]

========================================= ERRORS ==========================================
______________________________ ERROR at teardown of test_db _______________________________

    @fixture
    async def db() -> AsyncIterable[Database]:
        db: Database = Database("sqlite:///", force_rollback=True)
        await db.connect()
        try:
            yield db
        finally:
>           await db.disconnect()

tests/test_1.py:14: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.venv/lib/python3.12/site-packages/databases/core.py:141: in disconnect
    await self._global_transaction.__aexit__()
.venv/lib/python3.12/site-packages/databases/core.py:426: in __aexit__
    await self.rollback()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <databases.core.Transaction object at 0x75ea0da77470>

    async def rollback(self) -> None:
        async with self._connection._transaction_lock:
            assert self._connection._transaction_stack[-1] is self
            self._connection._transaction_stack.pop()
>           assert self._transaction is not None
E           AssertionError

.venv/lib/python3.12/site-packages/databases/core.py:473: AssertionError
```

The test not only fails but it also hangs.

Note that `databases` uses a contexVar

## Setup

To run this snippet, you need:

* databases[aiosqlite]
* pytest
* alt-pytest-asyncio

Can you help me solve that?

**Comment by delfick** [comment]
Hello,

I should have time on the weekend to take a look :)

**Comment by andredias** [comment]
Thanks!

**Comment by andredias** [comment]
The exact same issue was reported [here](https://github.com/encode/databases/issues/570#issuecomment-2061806278). There is another comment in the previous link that mentions [anyio pytest-plugin](https://github.com/encode/databases/issues/570#issuecomment-2062125145) as a possible solution.

I saw somewhere else about using [asyncio.Runner](https://docs.python.org/3.11/library/asyncio-runner.html#runner-context-manager) to keep the context among different tasks.

**Comment by delfick** [comment]
> I should have time on the weekend to take a look :)

Turns out that very much wasn't the case, we'll see if I do this weekend.

**Comment by delfick** [comment]
@andredias 
> I saw somewhere else about using [asyncio.Runner](https://docs.python.org/3.11/library/asyncio-runner.html#runner-context-manager) to keep the context among different tasks.

I tried using asyncio.Runner against the changes already in that PR and it got messy really quick. I'd need to do some very invasive surgery to this project to use that properly and I don't have the time atm.

Also, it would require the project being python 3.11 and above. I have put in that change into the PR anyway cause it removed one of the hacks the original attempt was doing.

Does that PR work for you now? https://github.com/delfick/alt-pytest-asyncio/pull/9

**Comment by delfick** [comment]
Fixed as part of the 0.8.0 release
